### PR TITLE
Add IBM X-Force Exchange Observable Analyzer

### LINKED ISSUE #139: XForce Intelligence Analyzer
Hello,

Is it possible to add XForce Intelligence (IBM XForce) to the list of APIs that can be used? They have a fully documented API for public use, along with limited free or paid API access. Some of the more useful ones (such as Botnet C2 categories) are not metered. 

https://api.xforce.ibmcloud.com/doc/
https://exchange.xforce.ibmcloud.com/
https://exchange.xforce.bmcloud.com/settings/api

"All" it requires is a valid API key and password, plus a valid GET request with a standardized format for the data. Example:

(Botnet C2 Servers - IPv4)
curl https://api.xforce.ibmcloud.com/xfti/bots/ipv4
{
  "FeedCategory": "Bots",
  "FeedType": "IPv4",
  "Version": "0000000010",
  "CreationDate": "2020-08-10T07:26:00.000Z",
  "IndicatorCount: "2",
  "data": ["127.0.0.1", "..."]
}

### PR DESCRIPTION
**Pull request by sp35**
Signed-off-by: Shubham Pandey <shubhampcpandey13@gmail.com>

# Description
Fixes #139 by @bowling
This adds an observable analyzer `XForceExchange`. Refer [139#issuecomment-798273872](https://github.com/intelowlproject/IntelOwl/issues/139#issuecomment-798273872)

## Type of change
- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] The pull request is for the branch develop
- [x] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
- [x] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/env_file_app_travis) and in the docs: [Installation](./Installation.md)
- [x] I have added tests in the [Tests](https://github.com/intelowlproject/IntelOwl/blob/master/tests) folder. 
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [x] I squashed the commits into a single one.
  
# Real World Example
image

Log Files - [xforce_logs.zip](https://github.com/intelowlproject/IntelOwl/files/6135666/xforce_logs.zip)

### CODE REVIEW

File: tests/test_observables.py [hunk 1]
```diff
@@ -555,6 +566,16 @@ def test_triage_submit(self, *args):
         ).start()
         self.assertEqual(report.get("success", False), True)
 
+    def test_xforce(self, mock_get=None, mock_post=None):
```
Comment (mlodic) [inline tests/test_observables.py:569]: we can avoid the duplication of testing code. Check how this is managed in `utils.py`.

You can create a new class `CommonTestCases_ip_url_hash` for this case and handle it similarly.

File: configuration/analyzer_config.json [hunk 2]
```diff
@@ -1334,6 +1334,23 @@
     },
     "soft_time_limit": 30
   },
+  "X-Force_Exchange": {
```
Comment (mlodic) [inline configuration/analyzer_config.json:1337]: combination of `-` and `_` could be a little tricky. Please remove them: `XForceExchange`

**Review by mlodic — CHANGES_REQUESTED** [review]
Thank you for your PR! I'll ask you 2 little changes

File: tests/test_observables.py (hunk 1 above)
Comment (sp35) [reply tests/test_observables.py]: That's so cool! I have changed this now :)

**Review by mlodic — APPROVED** [review]

**Comment by mlodic** [comment]
Great! Merged

**Pull request merged**